msglibrary/src/test/java/ingsw/group1/msglibrary/parsing/SMSParsingManagerTest.java

```diff
@@ -0,0 +1,347 @@
+package ingsw.group1.msglibrary.parsing;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+import ingsw.group1.msglibrary.RandomSMSPeerGenerator;
+import ingsw.group1.msglibrary.SMSMessage;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Class containing tests for {@link SMSParsingManager}.
+ * Robolectric is used to mock {@link android.content.SharedPreferences} automatically.
+ *
+ * @author Riccardo De Zen
+ */
+@Config(sdk = 28)
+@RunWith(RobolectricTestRunner.class)
+public class SMSParsingManagerTest {
+
+    private static final String EXAMPLE_NAME = "I'm a name";
+    private static final String ANOTHER_NAME = "Another name";
+    private static final SMSMessage EXAMPLE_MESSAGE = new SMSMessage(
+            new RandomSMSPeerGenerator().generateValidPeer(),
+            "I'm an example message"
+    );
+
+    private static String PREF_FILENAME;
+    private static String PREF_KEY_PREFIX;
+
+    private SMSParsingManager parsingManager;
+    private String prefKey;
+    private Context context;
+
+    /**
+     * Example parser to test the set and read feature in {@link SMSParsingManager}. It is
+     * relevant for this class to be a private child class because the way their name is written
+     * influences the reflection mechanisms used to instantiate the class.
+     */
+    public static class ExampleSMSParser implements SMSParser {
```

Comment:
`It is relevant for this class to be a private child class ..`
I see it's public, while `AnotherExampleSMSParser `is private, shouldn't this one be private too?